### ISSUE TITLE
Unexpected result when querying process instances in Flowable 7.0.1

### ISSUE THREAD

**Issue by pedrogalan** [flowable/flowable-engine#3827]
**Describe the bug**
Hi all,

I have found what it could be a bug in the recent `7.0.1` version. When using `HistoryService` to query process instances, it's using `or` instead of `and` for the query criteria. It was working well in `7.0.0`.

**Expected behavior**
Here is a test that, in my opinion, should be passing. I run a query with two criteria:
* `createdAfter(yesterday)` (always true)
* `processInstanceId == "NONEXISTENT_ID"` (always false)

I expect the `count()` to be `0`, as I expect the search to look for `criteria1 AND criteria2`

```
@Test
@Deployment(resources = ["processes/simple_process.bpmn20.xml"])
fun `counts returns 0 when filtering by unknown process instance id`() {
    // Define the filter
    val yesterday = OffsetDateTime.now().minusDays(1).toDate()
    val tasksQuery = historyService.createHistoricTaskInstanceQuery()
    tasksQuery.taskCreatedAfter(yesterday)
    tasksQuery.processInstanceId("NONEXISTENT_ID")

    // Create an instance
    runtimeService.startProcessInstanceByKey("simple_process")

    // Count the instances that match the filter
    val total = tasksQuery.count()

    // No instances are expected to match with id "NONEXISTENT_ID")
    assertEquals(0, total)
}

```

Looking at the logs, the SQL generated is as follows:

```
select count(distinct RES.ID_) 
from ACT_HI_TASKINST RES 
WHERE RES.PROC_INST_ID_ IN ( ? ) 
or RES.START_TIME_ > ?
```

In version `7.0.0` the SQL is as follows:

```
select count(distinct RES.ID_) 
from ACT_HI_TASKINST RES 
WHERE RES.PROC_INST_ID_ IN ( ? ) 
and RES.START_TIME_ > ?

```

(Notice `or` vs `and`)

**Code**
Looking at the [diff between 7.0.0 and 7.0.1](https://github.com/flowable/flowable-engine/compare/flowable-7.0.0...flowable-7.0.1), I see that [there has been a change in the file `HistoricTaskInstance.xml`](https://github.com/flowable/flowable-engine/blame/9aecece6e4ecdaa3c5368b1d8fe34bd2032a9190/modules/flowable-task-service/src/main/resources/org/flowable/task/service/db/mapping/entity/HistoricTaskInstance.xml#L1338) in the commit [`8e2be39`](https://github.com/flowable/flowable-engine/commit/8e2be39dca356bde5949fb07c706f068a35735ac) that might be related (some `and`'s have been replaced by `or`'s).

**Additional context**
I am using Flowable 7.0.0 with Spring Boot 3.2.1

**Comment by jbarrez** [comment]
Many thanks @pedrogalan for the excellent bug report and analysis!

Fixed in the referenced commit.

**Comment by jmoraleda** [comment]
I am impacted by this issue. Would you consider releasing a 7.0.2 version to maven with a fix? Alternatively, is there a way to downgrade a database (postgres) that was upgraded to the 7.0.1 back to the 7.0.0 format? Thank you

**Comment by tijsrademakers** [comment]
Hi, yes I think it's a good point to do a new 7.0.x release for this. We'll discuss it.

**Comment by jmoraleda** [comment]
Thank you. That's great. Do you have an ETA for this point release?

**Comment by jmoraleda** [comment]
<ping>

**Comment by jmoraleda** [comment]
> Hi, yes I think it's a good point to do a new 7.0.x release for this. We'll discuss it.

<ping>

**Comment by tijsrademakers** [comment]
Hi, this is still expected to be done, but we need to plan the release date.

**Comment by jmoraleda** [comment]
Thank you. I feel there is some urgency, as history search is currently broken for everyone who updated to the latest stable release.

**Comment by jmoraleda** [comment]
> Hi, this is still expected to be done, but we need to plan the release date.

Hello @tijsrademakers I wonder if you have had a chance to discuss a timeline for this. Thank you.

**Comment by vzickner** [comment]
@jmoraleda did you test the latest 7.1.0 release from last week?

**Comment by tijsrademakers** [comment]
Hi, as Valentin mentioned, we released 7.1.0 last week.

**Comment by jmoraleda** [comment]
Awesome! Thank you!
